ports/mimxrt/tusb_port.c

```diff
@@ -104,9 +109,31 @@ const uint16_t *tud_descriptor_string_cb(uint8_t index, uint16_t langid) {
         if (index >= sizeof(usbd_desc_str) / sizeof(usbd_desc_str[0])) {
             return NULL;
         }
+        if (index == USBD_STR_SERIAL)
+        {
+        uint8_t strs[DESC_STR_MAX];  
+        uint8_t uid[8];
+        mp_hal_get_unique_id(uid);
+                // convert it to hex string
+                for (int j = 0; j < 8; j++) {
+                    for (int k = 0; k < 2; k++) {
+                    strs[j * 2 + k] = (uid[j] & 0x0f) + 0x30;
+                    if (strs[j * 2 + k] > 0x39) {
+                        strs[j * 2 + k] += 7;
+                        }
+                    uid[j]>>=4;
+                }
+                }
+
+            for (len = 0; len < DESC_STR_MAX - 1 && strs[len]; ++len) {
+            desc_str[1 + len] = strs[len];
+        }
```

Comment:
DESC_STR_MAX is 20, but only the first 16 bytes of strs[] are set. So either change DESC_STR_MAX, or intialize [] to 0.